### PR TITLE
Rewrite: add webpack mdx

### PR DESCRIPTION
**Pull request by VirtualDOMinic**
First attempt at adding MDX support to rewrite.

Should we add `es/` and `dist/` to `.gitignore`?

### CODE REVIEW

**Comment by ryan-roemer** [comment]
Ignore es dist and lib. Only src should be in source

**Comment by ryan-roemer** [comment]
And they’re all git ignored correctly already. I’m not sure how they could get added in a branch?

**Comment by ryan-roemer** [comment]
Ah @kale-stew looks like task/rewrite branch is missing master's https://github.com/FormidableLabs/spectacle/blob/master/.gitignore Suggest we get that going in there sooner than later ;)

**Comment by VirtualDOMinic** [comment]
> And they’re all git ignored correctly already. I’m not sure how they could get added in a branch?

@ryan-roemer they're in the master `.gitignore` but not in the `task/rewrite` branch `.gitignore` that we're working off (yet)

**Comment by kale-stew** [comment]
My bad 🙂 Thank you for adding those

File: test.mdx [hunk 1]
```diff
@@ -0,0 +1,20 @@
+# Hello, _world_!
+
+Below is an example of JSX embedded in Markdown. <br /> **Kara and dom were successful!**
```
Comment (kale-stew) [inline test.mdx]: 🎉

File: package.json [hunk 2]
```diff
@@ -22,25 +23,41 @@
     "type": "git",
     "url": "https://github.com/FormidableLabs/spectacle.git"
   },
-  "dependencies": {},
+  "browserslist": "> 0.25%, not dead",
+  "dependencies": {
+    "webpack": "^4.33.0",
```
Comment (ryan-roemer) [inline package.json]: these should be devDeps

File: dist/index.html [hunk 3]
```diff
@@ -0,0 +1,9 @@
+<!DOCTYPE html>
```
Comment (kale-stew) [inline dist/index.html]: If `dist/` is being gitignored, should this still be showing up in the diff?

**Comment by ryan-roemer** [comment]
Thanks for ignore file @VirtualDOMinic !  i think the `dist/**` files still need to get manually removed too...

File: dist/index.html (hunk 3 above)
Comment (VirtualDOMinic) [reply dist/index.html]: Yep, they were still in the git cache, but I think my latest commit (`df726bd`) has potentially solved it! We'll get there in the end 😬

**Comment by VirtualDOMinic** [comment]
@ryan-roemer are you okay with us merging this in its current state? 😄

**Pull request merged**